### PR TITLE
Adds a recipe for the Double Fern

### PR DESCRIPTION
**Pull request by pmardle**
Adds the Double Fern as a craftable block - as it is requested by the builder when building some sandstone buildings.

Created by two normal ferns (Obtainable with shears on a fern/double fern) - Shapeless Recipe

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/util/RecipeHandler.java [hunk 1]
```diff
@@ -86,7 +86,10 @@ public static void init(final boolean enableInDevelopmentFeatures, final boolean
         GameRegistry.addRecipe(new ShapedOreRecipe(new ItemStack(ModBlocks.blockHutField, 1), 
                 " Y ", "X#X", " X ", 'X', WOODEN_STICK, '#', Items.LEATHER, 'Y', Blocks.HAY_BLOCK));
         GameRegistry.addRecipe(new ItemStack(Blocks.WEB, 1), "X X", " X ", "X X", 'X', Items.STRING);
-
+        
+        //Double Fern
+        GameRegistry.addRecipe(new ItemStack(Blocks.DOUBLE_PLANT, 1,3), "   ", " X ", " X ", 'X', Blocks.TALLGRASS, 1, 2);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/util/RecipeHandler.java]: MAJOR Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/coremod/util/RecipeHandler.java [hunk 2]
```diff
@@ -86,7 +86,10 @@ public static void init(final boolean enableInDevelopmentFeatures, final boolean
...
+        
         //enableInDevelopmentFeatures(enableInDevelopmentFeatures);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/util/RecipeHandler.java]: MINOR This block of commented-out lines of code should be removed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 30 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 27 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [RecipeHandler.java#L97](https://github.com/Minecolonies/minecolonies/blob/6c20c0434b18b136d33a84c123842afdb818377e/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L97): Remove this unused private "enableInDevelopmentFeatures" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [RecipeHandler.java#L116](https://github.com/Minecolonies/minecolonies/blob/6c20c0434b18b136d33a84c123842afdb818377e/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L116): Remove this unused private "addHutRecipe" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/6c20c0434b18b136d33a84c123842afdb818377e/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/6c20c0434b18b136d33a84c123842afdb818377e/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/6c20c0434b18b136d33a84c123842afdb818377e/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/6c20c0434b18b136d33a84c123842afdb818377e/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/6c20c0434b18b136d33a84c123842afdb818377e/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/6c20c0434b18b136d33a84c123842afdb818377e/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/6c20c0434b18b136d33a84c123842afdb818377e/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [entity](https://github.com/Minecolonies/minecolonies/blob/6c20c0434b18b136d33a84c123842afdb818377e/src/main/java/com/minecolonies/coremod/entity): Add a 'package-info.java' file to document the 'entity' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by marchermans** [comment]
This is not compiling. Please fix the compile error.

**Review by marchermans — CHANGES_REQUESTED** [review]
Not compiling. See the TeamCity build result.

**Comment by Raycoms** [comment]
:shipit: good to go now.
@OrionDevelopment should work now

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1281/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

File: src/main/java/com/minecolonies/coremod/util/RecipeHandler.java [hunk 3]
```diff
@@ -86,7 +86,10 @@ public static void init(final boolean enableInDevelopmentFeatures, final boolean
         GameRegistry.addRecipe(new ShapedOreRecipe(new ItemStack(ModBlocks.blockHutField, 1), 
                 " Y ", "X#X", " X ", 'X', WOODEN_STICK, '#', Items.LEATHER, 'Y', Blocks.HAY_BLOCK));
         GameRegistry.addRecipe(new ItemStack(Blocks.WEB, 1), "X X", " X ", "X X", 'X', Items.STRING);
-
+        
+        //Double Fern
+        GameRegistry.addShapelessRecipe(new ItemStack(Blocks.DOUBLE_PLANT, 1, 3), new ItemStack(Blocks.TALLGRASS, 1, 2), new ItemStack(Blocks.TALLGRASS, 1, 2));
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/util/RecipeHandler.java:91]: MAJOR Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/coremod/util/RecipeHandler.java [hunk 4]
```diff
@@ -86,7 +86,10 @@ public static void init(final boolean enableInDevelopmentFeatures, final boolean
...
+        
         //enableInDevelopmentFeatures(enableInDevelopmentFeatures);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/util/RecipeHandler.java:93]: MINOR This block of commented-out lines of code should be removed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 30 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 27 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [RecipeHandler.java#L97](https://github.com/Minecolonies/minecolonies/blob/c8a28702ec4fc6e90c785a37d9e56aa1f3b157d0/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L97): Remove this unused private "enableInDevelopmentFeatures" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [RecipeHandler.java#L116](https://github.com/Minecolonies/minecolonies/blob/c8a28702ec4fc6e90c785a37d9e56aa1f3b157d0/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L116): Remove this unused private "addHutRecipe" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/c8a28702ec4fc6e90c785a37d9e56aa1f3b157d0/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/c8a28702ec4fc6e90c785a37d9e56aa1f3b157d0/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/c8a28702ec4fc6e90c785a37d9e56aa1f3b157d0/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/c8a28702ec4fc6e90c785a37d9e56aa1f3b157d0/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/c8a28702ec4fc6e90c785a37d9e56aa1f3b157d0/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/c8a28702ec4fc6e90c785a37d9e56aa1f3b157d0/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/c8a28702ec4fc6e90c785a37d9e56aa1f3b157d0/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [entity](https://github.com/Minecolonies/minecolonies/blob/c8a28702ec4fc6e90c785a37d9e56aa1f3b157d0/src/main/java/com/minecolonies/coremod/entity): Add a 'package-info.java' file to document the 'entity' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

**Pull request merged**